prediction_market_agent/agents/microchain_agent/blockchain/contract_class_converter.py

```diff
@@ -0,0 +1,184 @@
+import json
+from collections import defaultdict
+from typing import Any, Tuple
+
+import requests
+from eth_typing import ChecksumAddress
+from loguru import logger
+from microchain import Function
+from prediction_market_agent_tooling.gtypes import ABI
+from prediction_market_agent_tooling.tools.contract import ContractOnGnosisChain
+from web3 import Web3
+
+from prediction_market_agent.agents.microchain_agent.blockchain.code_interpreter import (
+    CodeInterpreter,
+    FunctionSummary,
+    Summaries,
+)
+from prediction_market_agent.agents.microchain_agent.blockchain.models import (
+    AbiItemStateMutabilityEnum,
+    AbiItemTypeEnum,
+    ABIMetadata,
+)
+from prediction_market_agent.utils import APIKeys
+
+TYPE_MAPPING: dict[str, Tuple[str, Any]] = {
+    "address": (
+        "str",
+        Web3.to_checksum_address("0xe91d153e0b41518a2ce8dd3d7944fa863463a97d"),
+    ),  # Solidity address to Python string
+    "uint": ("int", 1),  # Solidity unsigned integer to Python int
+    "uint8": ("int", 1),  # Solidity unsigned integer to Python int
+    "uint256": ("int", 1),  # Solidity unsigned integer (256-bit) to Python int
+    "int": ("int", 1),  # Solidity integer to Python int
+    "bool": ("bool", True),  # Solidity boolean to Python bool
+    "string": ("str", "test string"),
+    # Add more mappings as needed
+}
+
+
+class FunctionWithKeys(Function):
+    def __init__(self, keys: APIKeys) -> None:
+        self.keys = keys
+        super().__init__()
+
+
+class ClassFactory:
+    @staticmethod
+    def create_class(
+        class_name: str,
+        base_classes: Tuple[type],
```

Comment:
Looks like you don't need a Tuple here at all, since the only place this is called is:
```
        dynamic_class = ClassFactory().create_class(
            abi_item.name.title(), (base,), attributes
        )
```

Reply:
@gabrielfior ?